src/scene/sprite-atlas.js

```diff
@@ -0,0 +1,34 @@
+pc.extend(pc, function () {
+    'use strict';
+
+    /**
+    * @name pc.SpriteAtlas
+    * @class Represents the resource of a sprite atlas asset.
+    * @property {pc.Texture} texture The texture atlas.
+    * @property {Object} frames Contains frames which define portions of the texture atlas.
+    * @example
+    * var atlas = new pc.SpriteAtlas();
+    * atlas.frames = {
+    *   '0': {
+    *       // rect has u, v, width and height which are 0-1 values
+    *       rect: new pc.Vec4(0, 0, 0.5, 0.5),
+    *       // pivot has x, y values between 0-1 which define the point
+    *       // within the frame around which rotation and scale is calculated
+    *       pivot: new pc.Vec2(0.5, 0.5)
+    *   },
+    *   '1': {
+    *       rect: new pc.Vec4(0.5, 0.5, 0.5, 0.5),
+    *       pivot: new pc.Vec2(0.5, 0.5)
+    *   },
+    *   ...
+    * };
+    */
+    var SpriteAtlas = function () {
+        this.texture = null;
+        this.frames = null;
```

Comment:
Should this be an empty array?

Reply:
It's a dictionary. We could initialize it as a dictionary although that's gonna be garbage collected once we load the real frames unless we always modify the same object...